platform-services.md

```diff
@@ -0,0 +1,215 @@
++ [`input_event_mojom`](http://docs.flutter.io/flutter/input_event_mojom/input_event_mojom-library.html)
++ [`media_mojom`](http://docs.flutter.io/flutter/media_mojom/media_mojom-library.html)
++ [`path_provider_mojom`](http://docs.flutter.io/flutter/path_provider_mojom/path_provider_mojom-library.html)
++ [`pointer_mojom`](http://docs.flutter.io/flutter/pointer_mojom/pointer_mojom-library.html)
++ [`raw_keyboard_mojom`](http://docs.flutter.io/flutter/raw_keyboard_mojom/raw_keyboard_mojom-library.html)
++ [`semantics_mojom`](http://docs.flutter.io/flutter/semantics_mojom/semantics_mojom-library.html)
++ [`system_chrome_mojom`](http://docs.flutter.io/flutter/system_chrome_mojom/system_chrome_mojom-library.html)
++ [`system_sound_mojom`](http://docs.flutter.io/flutter/system_sound_mojom/system_sound_mojom-library.html)
+
+### Platform and Third-party Services for Flutter {#platform-and-third-party-services-for-flutter}
+
+To access APIs that are native to the iOS or Android mobile platform, Flutter provides the
+`FlutterView` interface. The `FlutterView` interface allows you to embed a bi-directional
+asynchronous messaging component in your platform code to send and receive native service calls
+in a transparent way. This flexible approach lets you create cross-platform mobile apps that can
+connect to platform-specific and third-party services on iOS and Android devices, while sharing a
+common user interface (UI) codebase in Dart.
+
+You can build your own integrations with platform-specific services and third-party SDKs. Ideally,
+the services that you invoke using the `FlutterView` interface should return non-graphical,
+data-only information. Examples include:
+
+*   Geolocation
+*   Contacts
+*   File system
+*   Network information
+*   Sensors
+
+You can also use the `FlutterView` interface to invoke APIs from third-party SDKs that run on iOS
+and Android devices.
+
+To use the `FlutterView` interface, you must add a platform-specific Flutter component into the
+platform’s view hierarchy. For example, on iOS, you add a `FlutterViewController` object to
+interact with a `UIView`.  Similarly, on Android, you add a `FlutterView` object to your app’s
+`android.view.View` hierarchy. Your app can then communicate between the Dart and iOS or
+Android-specific parts of the app by using a simple message-passing interface.
+
+### Message passing {#message-passing}
+
+Flutter offers a simple message-passing interface between your platform-specific code and the
+Dart code. The message passing interface is asynchronous, bi-directional, and supports optional
+reply routing. Messages consist of strings encoded in JSON format, and these is no facility for
+transferring handles to other objects along with messages. You are responsible for encoding and
+decoding messages sent through this mechanism.
+
+## Get the `hello_services` sample {#the-hello_services-sample}
```

Comment:
It's an "example" rather than a "sample".  I'm not sure what the difference between those words is, but it's in a directory called "examples" rather than a directory called "samples".